opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1194,23 +1194,23 @@ namespace Opm
 
         // COMPDAT handling
         const auto& connectionSet = well_ecl_.getConnections();
+        CheckDistributedWellConnections checker(well_ecl_, parallel_well_info_);
         for (size_t c=0; c<connectionSet.size(); c++) {
             const auto& connection = connectionSet.get(c);
+            const int i = connection.getI();
+            const int j = connection.getJ();
+            const int k = connection.getK();
+
+            const int* cpgdim = cart_dims;
+            const int cart_grid_indx = i + cpgdim[0]*(j + cpgdim[1]*k);
+            const int cell = cartesian_to_compressed[cart_grid_indx];
```

Comment:
Same here.  There's no need to perform this arithmetic when the value is already stored in the `Connection` object.